packages/dd-trace/src/llmobs/sdk.js

```diff
@@ -355,6 +367,10 @@ class LLMObs extends NoopLLMObs {
     }
   }
 
+  _getParent (parent, childOf) {
+    return childOf instanceof Span && LLMObsTagger.tagMap.has(childOf) ? childOf : parent
```

Comment:
I think this condition would subtly hide programming errors with the type of `childOf`. I'd rather we just take the value of `childOf` if specified and let it fail wherever it might if it's the wrong type or fail hard up front in these cases rather than subtly falling back to `parent` here.